packages/worker-threads-pool/src/AggregatedPool.ts

```diff
@@ -0,0 +1,71 @@
+import type { AbortSignal } from "abort-controller";
+import type { Readable } from "stream";
+import type { Worker, WorkerOptions } from "worker_threads";
+import type { Pool } from "./types/Pool";
+import type { Logger } from "@lage-run/logger";
+
+import { WorkerPool } from "./WorkerPool";
+
+interface AggregatedPoolOptions {
+  groupBy: (data: any) => string;
+  maxWorkersByGroup: Map<string, number>;
+  maxWorkers: number;
+  script: string;
+  workerOptions?: WorkerOptions;
+  logger: Logger;
+}
+
+export class AggregatedPool implements Pool {
+  readonly groupedPools: Map<string, WorkerPool> = new Map();
+  readonly defaultPool: WorkerPool | undefined;
+
+  constructor(private options: AggregatedPoolOptions) {
+    const { maxWorkers, maxWorkersByGroup, script, workerOptions } = options;
+
+    let totalGroupedWorkers = 0;
+    for (const [group, groupMaxWorkers] of maxWorkersByGroup.entries()) {
+      const pool = new WorkerPool({ maxWorkers: groupMaxWorkers, workerOptions, script });
+      this.groupedPools.set(group, pool);
+      totalGroupedWorkers += groupMaxWorkers;
+    }
+
+    if (totalGroupedWorkers > maxWorkers) {
+      throw new Error(
+        `Total maxWorkers (${totalGroupedWorkers}) configured across all groups exceeds concurrency (${maxWorkers}). Hint: use percent of concurrency to configure the maxWorkers like this: { maxWorkers: "50%" }`
+      );
```

Comment:
Didn't you just outlaw strings in the config, or is that a different setting?

Reply:
(if you support % it must be a different setting)

Reply:
yes, added some